Common/util/cmdlineopts.cpp

```diff
@@ -0,0 +1,91 @@
+//=============================================================================
+//
+// Adventure Game Studio (AGS)
+//
+// Copyright (C) 1999-2011 Chris Jones and 2011-20xx others
+// The full list of copyright holders can be found in the Copyright.txt
+// file, which is part of this source code distribution.
+//
+// The AGS source code is provided under the Artistic License 2.0.
+// A copy of this license can be found in the file License.txt and at
+// http://www.opensource.org/licenses/artistic-license-2.0.php
+//
+//=============================================================================
+
+#include "core/platform.h"
+#include "util/string_compat.h"
+#include "util/cmdlineopts.h"
+
+namespace AGS
+{
+namespace Common
+{
+
+namespace CmdLineOpts
+{
+
+ParseResult Parse(int argc, char* argv[], const std::set<String> &opt_params_with_values)
```

Comment:
Ah, I did not notice this earlier, but the `argv` parameter might also be `const`, that would also help with tests where you are forced to cast `const char*` to `char*`.

This is a separate issue, but we also have this mistake in the engine code.

Reply:
this one has to match the signature of our main, which doesn't have const.

Reply:
You may convert non const char* array to const char* array, see my "edit", I've tested that in the engine code.

Reply:
Thanks for the `const char *const argv[]`, I couldn't figure this! Did the change here:

https://github.com/adventuregamestudio/ags/pull/1370/commits/a922cee7f09b669c4b0073e75061cc14048ca579#diff-8b3b8298bbda3a3eee264e0d8994d6294883e3edc45b3b1981a24c8663cf504fL43-R43